provider_settings.json

```diff
@@ -1,4 +1,8 @@
 [
+    {
+      "name": "builtin",
```

Comment:
should I just hardcode this in the engine?

Reply:
I guess we want to keep location configurable so maybe not

Reply:
what would the location be in this case?

Reply:
The root of the code base, unless we want to assume it's just going to be `./`

Reply:
yeah I wonder if anything in builtin is a provider that we provide and is always present